Fix active storage firefox

lazaronixon opened the pull request:
### Summary
Direct upload on firefox is not working because it can’t handle click
events on “input[type=submit]”.
After a successful upload we can see on console a xml parse error, it
happens because AWS S3 success body response is empty, so changing the
reponseType suppress this error.

### Other Information
Xml parse error on console is not a problem, it works without problem.
If i use button.focus() before button.click() it works too, but on next lines of code we have a condition to insert a hidden input[submit] tag and again firefox cant click hidden input[submit] tags

starknx commented on activestorage/app/javascript/activestorage/blob_upload.js:
```diff
@@ -7,6 +7,7 @@ export class BlobUpload {
 
     this.xhr = new XMLHttpRequest
     this.xhr.open("PUT", url, true)
+    this.xhr.responseType = "text"
```
Suppress xml parser error on console

starknx commented on activestorage/app/javascript/activestorage/ujs.js:
```diff
@@ -42,29 +41,12 @@ function handleFormSubmissionEvent(event) {
-  if (button) {
-    const { disabled } = button
-    button.disabled = false
-    button.click()
```
button.focus() before clicks solves here too

starknx commented on activestorage/app/javascript/activestorage/ujs.js:
```diff
@@ -42,29 +41,12 @@ function handleFormSubmissionEvent(event) {
-    button.type = "submit"
-    button.style = "display:none"
-    form.appendChild(button)
-    button.click()
```
button.focus() not solves here because display:none

javan commented on activestorage/app/javascript/activestorage/ujs.js:
```diff
@@ -42,29 +41,12 @@ function handleFormSubmissionEvent(event) {
         inputs.forEach(enable)
       } else {
-        submitForm(form)
+        form.submit()
```
This will prevent submitting "remote" forms via XHR, unfortunately, and do a full HTTP request instead. See: https://github.com/rails/rails/blob/5af7d475c8cd7d3dbffca7452f3dcb221896e8e5/actionview/app/assets/javascripts/rails-ujs/features/method.coffee#L21-L23 https://github.com/rails/rails/blob/5af7d475c8cd7d3dbffca7452f3dcb221896e8e5/actionview/app/assets/javascripts/rails-ujs/features/method.coffee#L32

Will need to explore using the `button.focus()` solution you suggested below.

starknx commented:
Seems focus solved the problem... I tested second case without focus and it works too...

javan commented:
Can you [squash your commits](http://edgeguides.rubyonrails.org/contributing_to_ruby_on_rails.html#squashing-commits), please?

lazaronixon committed:
Fix activestorage direct upload on firefox

Direct upload on firefox is not working because it can’t handle click
events on “input[type=submit]”.
After a successful upload we can see on console a xml parse error, it
happens because AWS S3 success body response is empty, so changing the
reponseType suppress this error.

javan commented:
Thank you!
